lib/charms/layer/apache_bigtop_base.py

```diff
@@ -80,18 +88,27 @@ def render_hiera_yaml(self):
         hiera_yaml = yaml.load(hiera_src.text())
 
         # set the datadir
-        hiera_yaml[':yaml'][':datadir'] = str(self.site_yaml.dirname())
+        hiera_yaml[':yaml'][':datadir'] = str(self.hieradata_path)
+
+        # insert <charm>.yaml into the hierarchy, after site.yaml
+        hiera_yaml[':hierarchy'].insert(1, self.charm_yaml)
```

Comment:
This needs to check if it's already in the hierarchy, on the off chance it gets called more than once.